File: Counting Bits/counting bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        result = []
        for i in range(n+1):
            result.append(self.getNoOfSetBits(i))
        return result
    
    def getNoOfSetBits(self, num: int) -> int:
        count = 0
        for i in range(32):
            if ((num & (1 << i)) > 0):
                count += 1
        return count

File: Counting Bits/counting bits.java
class Solution {
    public int[] countBits(int n) {
        int[] result = new int[n + 1];
        
        for(int i = 0; i <= n; i++) {
            result[i] = getNoOfSetBits(i);
        }
        
        return result;
        
    }
    
    int getNoOfSetBits(int num) {
        int count = 0;
        for(int i = 0; i <= 31; i++) {
            if ((num & (1 << i)) > 0) {
                count++;
            }
        }
        return count;
    }
}